
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>1.3. omics</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/bootstrap.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />

  
  <link href="../../_static/vendor/fontawesome/6.5.2/css/all.min.css?digest=dfe6caa3a7d634c4db9b" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="../../_static/vendor/fontawesome/6.5.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=eba8b062" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.8ecb98da25f57f5357bf6f572d296f466b2cfe2517ffebfabe82451661e28f02.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b" />
  <script src="../../_static/vendor/fontawesome/6.5.2/js/all.min.js?digest=dfe6caa3a7d634c4db9b"></script>

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'notes/research_ovws/ovw_omics';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="1.4. llms" href="ovw_llms.html" />
    <link rel="prev" title="1.2. disentanglement" href="ovw_disentanglement.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-primary-sidebar-checkbox"/>
  <label class="overlay overlay-primary" for="pst-primary-sidebar-checkbox"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          id="pst-secondary-sidebar-checkbox"/>
  <label class="overlay overlay-secondary" for="pst-secondary-sidebar-checkbox"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo.png" class="logo__image only-light" alt=" - Home"/>
    <script>document.write(`<img src="../../_static/logo.png" class="logo__image only-dark" alt=" - Home"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item">

 <script>
 document.write(`
   <button class="btn search-button-field search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
    <span class="search-button__default-text">Search</span>
    <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
   </button>
 `);
 </script></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../../intro.html">
                    overview 👋
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
<li class="toctree-l1 current active has-children"><a class="reference internal" href="research_ovws.html">1. research_ovws</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="ovw_transfer_learning.html">1.1. transfer learning</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_disentanglement.html">1.2. disentanglement</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">1.3. omics</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_llms.html">1.4. llms</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_complexity.html">1.5. complexity</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_interesting_science.html">1.6. interesting science</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_dl_theory.html">1.7. dl theory</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_scat.html">1.8. scattering transform</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_ml_medicine.html">1.9. ml in medicine</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_causal_inference.html">1.10. causal inference</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_uncertainty.html">1.11. uncertainty</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_interp.html">1.12. interpretability</a></li>
<li class="toctree-l2"><a class="reference internal" href="ovw_generalization.html">1.13. generalization</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../cs/cs.html">2. cs</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../cs/retrieval.html">2.1. info retrieval</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/data_structures.html">2.2. data structures</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/languages.html">2.3. languages</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/software.html">2.4. software engineering</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/quantum.html">2.5. quantum</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/algo.html">2.6. algorithms</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/graphs.html">2.7. graphs</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/os.html">2.8. os</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/arch.html">2.9. architecture</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/reproducibility.html">2.10. reproducibility</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cs/comp_theory.html">2.11. cs theory</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../math/math.html">3. math</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../math/differential_equations.html">3.1. differential equations</a></li>
<li class="toctree-l2"><a class="reference internal" href="../math/proofs.html">3.2. proofs</a></li>
<li class="toctree-l2"><a class="reference internal" href="../math/analysis.html">3.3. real analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../math/linear_algebra.html">3.4. linear algebra</a></li>
<li class="toctree-l2"><a class="reference internal" href="../math/signals.html">3.5. signals</a></li>
<li class="toctree-l2"><a class="reference internal" href="../math/optimization.html">3.6. optimization</a></li>
<li class="toctree-l2"><a class="reference internal" href="../math/calculus.html">3.7. calculus</a></li>
<li class="toctree-l2"><a class="reference internal" href="../math/chaos.html">3.8. chaos</a></li>
<li class="toctree-l2"><a class="reference internal" href="../math/math_basics.html">3.9. math basics</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../stat/stat.html">4. stat</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../stat/graphical_models.html">4.1. graphical models</a></li>
<li class="toctree-l2"><a class="reference internal" href="../stat/data_analysis.html">4.2. data analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="../stat/testing.html">4.3. testing</a></li>
<li class="toctree-l2"><a class="reference internal" href="../stat/causal_inference.html">4.4. causal inference</a></li>
<li class="toctree-l2"><a class="reference internal" href="../stat/info_theory.html">4.5. info theory</a></li>
<li class="toctree-l2"><a class="reference internal" href="../stat/linear_models.html">4.6. linear models</a></li>
<li class="toctree-l2"><a class="reference internal" href="../stat/time_series.html">4.7. time series</a></li>
<li class="toctree-l2"><a class="reference internal" href="../stat/game_theory.html">4.8. game theory</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../ml/ml.html">5. ml</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../ml/kernels.html">5.1. kernels</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/nlp.html">5.2. nlp</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/comp_vision.html">5.3. computer vision</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/structure_ml.html">5.4. structure learning</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/classification.html">5.5. classification</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/unsupervised.html">5.6. unsupervised</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/deep_learning.html">5.7. deep learning</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/feature_selection.html">5.8. feature selection</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/learning_theory.html">5.9. learning theory</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ml/evaluation.html">5.10. evaluation</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../ai/ai.html">6. ai</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../ai/llms.html">6.1. llms</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/search.html">6.2. search</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/decisions_rl.html">6.3. decisions, rl</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/fairness_sts.html">6.4. fairness, sts</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/cogsci.html">6.5. cognitive science</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/ai_futures.html">6.6. ai futures</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/logic.html">6.7. logic</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/philosophy.html">6.8. philosophy</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/psychology.html">6.9. psychology</a></li>
<li class="toctree-l2"><a class="reference internal" href="../ai/knowledge_rep.html">6.10. representations</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../neuro/neuro.html">7. neuro</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../neuro/disease.html">7.1. disease</a></li>
<li class="toctree-l2"><a class="reference internal" href="../neuro/brain_basics.html">7.2. brain basics</a></li>
<li class="toctree-l2"><a class="reference internal" href="../neuro/vissci.html">7.3. vision</a></li>
<li class="toctree-l2"><a class="reference internal" href="../neuro/comp_neuro.html">7.4. comp neuro</a></li>
<li class="toctree-l2"><a class="reference internal" href="../neuro/sensory_input.html">7.5. sensory input</a></li>
<li class="toctree-l2"><a class="reference internal" href="../neuro/memory.html">7.6. memory</a></li>
<li class="toctree-l2"><a class="reference internal" href="../neuro/motor.html">7.7. motor system</a></li>
<li class="toctree-l2"><a class="reference internal" href="../neuro/development.html">7.8. development</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">


<a href="https://github.com/csinva/csinva.github.io" target="_blank"
   class="btn btn-sm btn-source-repository-button"
   title="Source repository"
   data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>

</a>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/notes/research_ovws/ovw_omics.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<script>
document.write(`
  <button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="theme-switch fa-solid fa-sun fa-lg" data-mode="light"></i>
    <i class="theme-switch fa-solid fa-moon fa-lg" data-mode="dark"></i>
    <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"></i>
  </button>
`);
</script>


<script>
document.write(`
  <button class="btn btn-sm pst-navbar-icon search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
  </button>
`);
</script>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>omics</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#overview">1.3.1. overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#vocabulary">1.3.1.1. vocabulary</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#data">1.3.1.2. data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#code">1.3.1.3. code</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#rna-structure-prediction">1.3.2. rna structure prediction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#rna-basics">1.3.2.1. rna basics</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#algorithms">1.3.2.2. algorithms</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#d-rna-structure-prediction">1.3.2.3. 3d rna structure prediction</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#protein-structure-prediction-dna">1.3.3. protein structure prediction (dna)</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#protein-basics">1.3.3.1. protein basics</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#algorithm-basics">1.3.3.2. algorithm basics</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#deep-learning-papers">1.3.3.3. deep-learning papers</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#covid">1.3.3.4. covid</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#tips-using-viruses-for-treatment">1.3.4. TIPs - using viruses for treatment</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#random-papers">1.3.5. random papers</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="omics">
<h1><span class="section-number">1.3. </span>omics<a class="headerlink" href="#omics" title="Link to this heading">#</a></h1>
<section id="overview">
<h2><span class="section-number">1.3.1. </span>overview<a class="headerlink" href="#overview" title="Link to this heading">#</a></h2>
<section id="vocabulary">
<h3><span class="section-number">1.3.1.1. </span>vocabulary<a class="headerlink" href="#vocabulary" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><em>oligonucleotide</em> = <em>oligo</em> = short single strands of synthetic DNA or <em>RNA</em></p></li>
</ul>
</section>
<section id="data">
<h3><span class="section-number">1.3.1.2. </span>data<a class="headerlink" href="#data" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="http://ndbserver.rutgers.edu/">nucleic acid database</a> (NDB)</p></li>
<li><p><a class="reference external" href="https://www.rcsb.org/">protein databank</a> (PDB)</p></li>
<li><p>rna</p>
<ul>
<li><p>~~<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC165459/">RNABase: an annotated database of RNA structures</a>~~ (2003) - no longer maintained</p></li>
<li><p><a class="reference external" href="https://www.pnas.org/content/106/1/97">Accurate SHAPE-directed RNA structure determination | PNAS</a> (deigan et al. 2009) - chemical probing</p></li>
</ul>
</li>
</ul>
</section>
<section id="code">
<h3><span class="section-number">1.3.1.3. </span>code<a class="headerlink" href="#code" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>rna</p>
<ul>
<li><p><a class="reference external" href="https://github.com/bgruening/galaxy-rna-workbench">Galaxy RNA workbench</a> - many tools including alignment, annotation, interaction</p></li>
<li><p><a class="reference external" href="https://github.com/ViennaRNA/ViennaRNA">viennaRNA</a> - incorporating constraints into predictions</p></li>
<li><p>neural nets</p>
<ul>
<li><p><a class="reference external" href="https://github.com/jaswindersingh2/SPOT-RNA">spot rna</a> / <a class="reference external" href="https://github.com/jaswindersingh2/SPOT-RNA2">spot rna2</a></p>
<ul>
<li><p>uses an Ensemble of Two-dimensional Deep Neural Networks and Transfer Learning</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://github.com/keio-bioinformatics/mxfold2/">mxfold2</a> (sato et al. 2021)</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</section>
<section id="rna-structure-prediction">
<h2><span class="section-number">1.3.2. </span>rna structure prediction<a class="headerlink" href="#rna-structure-prediction" title="Link to this heading">#</a></h2>
<section id="rna-basics">
<h3><span class="section-number">1.3.2.1. </span>rna basics<a class="headerlink" href="#rna-basics" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>rna does many things</p>
<ul>
<li><p>certain structures of RNA lend themselves to catalytic activities</p></li>
<li><p>tRNA/mRNA convert DNA into proteins</p></li>
<li><p>RNA also serves as the information storage and replication agent in some viruses</p></li>
</ul>
</li>
<li><p>rna components</p>
<ul>
<li><p>4 bases: adenine-uracil (instead of dna’s thymine), cytosine-guanine</p></li>
<li><p>ribose sugar in RNA is more flexible than DNA</p></li>
</ul>
</li>
<li><p>RNA World Hypothesis (Walter Gilbert, 1986) - suggests RNA was precursor to modern life</p>
<ul>
<li><p>later dna stepped in for info storage (more stable) and proteins took over for catalyzing reactions</p></li>
</ul>
</li>
<li><p>rna structure</p>
<ul>
<li><p><img alt="rna_structure" src="../../_images/rna_structure.png" /></p>
<ul>
<li><p>primary - sequence in which the bases are aligned - relatively simple, comes from sequencing</p></li>
<li><p><strong>secondary</strong> - 2d analysis of hydrogen bonds between rna parts (double-strands, hairpins, loops) - most of stabilizing free energy comes from here (unlike proteins, where tertiary is most important)</p>
<ul>
<li><p>most work on “RNA folding” predicts secondary structure from primary structure</p></li>
<li><p>a lot of this comes from hydrogen bonds between pairs (watson-crick edge)</p>
<ul>
<li><p>other parts of the pairs (e.g. the Hoogsteen- / CH-edge and the sugar edge) can also form bonds</p></li>
</ul>
</li>
</ul>
</li>
<li><p>tertiary - complete 3d structure (e.g. bends, twists)</p></li>
</ul>
</li>
<li><p><img alt="" src="../../_images/rna_secondary.png" /></p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://en.wikipedia.org/wiki/RNA-Seq">RNA-Seq - Wikipedia</a> - RNA-Seq uses next-generation sequencing (NGS) to reveal the presence and quantity of RNA in a biological sample at a given moment, analyzing the continuously changing cellular transcriptome</p></li>
</ul>
</section>
<section id="algorithms">
<h3><span class="section-number">1.3.2.2. </span>algorithms<a class="headerlink" href="#algorithms" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://ocw.mit.edu/ans7870/6/6.047/f15/MIT6_047F15_Compiled.pdf">computational bio book ch 10</a> (kellis, 2016)</p>
<ul>
<li><p>2 main approaches to rna folding (i.e. predicting rna structure):</p>
<ul>
<li><p>(1) thermodynamic stability of the molecule</p></li>
<li><p>(2) probabilistic models</p></li>
<li><p>note: can use evolutionary (i.e. phylogenetic) data to improve either of these</p>
<ul>
<li><p>some RNA changes still result in similar structures</p></li>
<li><p>consistent mutations - something mutates but structure doesn’t change (e.g. AU -&gt; G)</p></li>
<li><p>compensatory mutations - two mutations but structure doesn’t change (e.g. AU -&gt; CU -&gt; CG)</p></li>
<li><p>incorporate similarities to other RNAs along with something like zuker algorithm</p></li>
</ul>
</li>
</ul>
</li>
<li><p>thermodynamic stability - uses more domain knowledge</p>
<ul>
<li><p>dynamic programming approach: given energy value for each pair, minimize total energy by pairing up appropriate base pairs</p>
<ul>
<li><p><strong>assumption</strong>: no crossings - given a subsequence <span class="math notranslate nohighlight">\([i,j]\)</span>, there is either no edge connecting to the ith base (meaning it is unpaired) or there is some edge connecting the ith base to the kth base where <span class="math notranslate nohighlight">\(i &lt; k \leq j\)</span> (meaning the ith base is paired to the kth base)</p>
<ul>
<li><p>this induces a tree-structure to theh problem</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.pnas.org/content/77/11/6309.short">nussinov algorithm</a> (1978)</p>
<ul>
<li><p>ignores stacking interactions between neighboring pairs (i.e. assumes there are no pseudo-knots)</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://academic.oup.com/nar/article-abstract/9/1/133/1043226">zuker algorithm</a> (1981) - includes stacking energies</p></li>
</ul>
</li>
</ul>
</li>
<li><p>probabilistic approach - uses more statistical likelihood</p>
<ul>
<li><p>stochastic context-free grammer (SCFG) is like an extension of HMM that incorporates some RNA constraints (e.g. bonds must happen between pairs)</p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0168165617315195">Recent advances in RNA folding - ScienceDirect</a> (fallmann et al. 2017)</p>
<ul>
<li><p>algorithmic advances</p>
<ul>
<li><p>algorithmic constraints</p>
<ul>
<li><p>locality - restrict maximal span of base pairs</p></li>
<li><p>add in bonus energies as hard or soft constraints</p></li>
</ul>
</li>
<li><p>rna-rna and rna-protein interactions - there are different algorithms for telling how 2 things will interact</p></li>
<li><p>newer algorithms take into account some aspects of tertiary structure to predict secondary structure (e.g. motifs, nonstandard base pairs, pseudo-knots)</p></li>
</ul>
</li>
<li><p>representation</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">...((((((((...)).</span> <span class="pre">.)).</span> <span class="pre">.((.((...)).</span> <span class="pre">)))))).</span></code> “dot-parenthesis” notation - opening and closing represent bonded pairs</p></li>
</ul>
</li>
<li><p>evaluation</p>
<ul>
<li><p>computing alignments is non-obvious in these representations</p></li>
<li><p>centroids - structures with a minimum distance to all other structures in the ensemble of possible structures</p></li>
<li><p>consensus structures - given a good alignment of a collection of related RNA structures, can compute their consensus structure, (i.e., a set of base pairs at corresponding alignment positions)</p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2982177/">Folding and Finding RNA Secondary Structure</a> (matthews et al. 2010)</p></li>
<li><p>new work drops the assumption of no knots (e.g. crossings)</p>
<ul>
<li><p>project to topology (e.g. low/high genus) - e.g. projecting to a torus can remove crossings but still allow us to use dynamic programming</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.nature.com/articles/s41467-021-21194-4">RNA secondary structure prediction using deep learning with thermodynamic integration</a> (sato et al. 2021)</p>
<ul>
<li><p>RNA-folding scores learnt using a DNN are integrated together with Turner’s nearest-neighbor free energy parameters</p>
<ul>
<li><p>DNN predicts scores that are fed into zuker-style dynamic programming</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
<section id="d-rna-structure-prediction">
<h3><span class="section-number">1.3.2.3. </span>3d rna structure prediction<a class="headerlink" href="#d-rna-structure-prediction" title="Link to this heading">#</a></h3>
<ul>
<li><p>Geometric deep learning of RNA structure  (<a class="reference external" href="https://www.science.org/doi/10.1126/science.abe5650">townshend, …, dror 2021</a> - atomic ai + stanford, <em>science</em>)</p>
<ul class="simple">
<li><p>uses neural network as scoring mode (called ARES = Atomic Rotationally Equivariant Scorer)</p>
<ul>
<li><p>this gives energy function for (primary sequence, structure) pair</p></li>
</ul>
</li>
<li><p>architecture</p>
<ul>
<li><p>captures rotational + translational symmetries</p></li>
</ul>
</li>
<li><p>startup <a class="reference external" href="https://atomic.ai/">atomic ai</a> works on this</p></li>
<li><p>trained with only 18 known RNA structures</p></li>
<li><p>uses only atomic coordinates as inputs and incorporates no RNA-specific information</p></li>
</ul>
</li>
<li><p>Review: RNA 3D Structure Prediction Using Coarse-Grained Model (<a class="reference external" href="https://www.frontiersin.org/articles/10.3389/fmolb.2021.720937/full">li &amp; chen, 2021</a>)</p>
<ul>
<li><p>all-atom - more precise, each nucleotide models ~20 heavy atoms + ~10 hydrogen atoms - relatively rare</p></li>
<li><p>coarse-grained - less precise</p></li>
<li><p>steps</p>
<ol class="arabic simple">
<li><p>input - RNA sequence + optinal secondary structure + more (e.g. distance between some atom pairs)</p></li>
<li><p>sampling - includes discriminator (=scoring function = (potential) energy function = force field) and generator</p></li>
</ol>
<ul class="simple">
<li><p>generator proposes new structures (e.g. MCMC)</p></li>
<li><p>discriminator decides which is the most stable - 3 types of energy to consider</p></li>
</ul>
<ol class="arabic simple" start="3">
<li><p>output - sometimes pick lowest energy structure, somestimes cluster low-energy clusters and pick centroid</p></li>
<li><p>all-atom structure reconstruction - the fragment matching algorithm (<a class="reference external" href="https://academic.oup.com/bioinformatics/article-abstract/25/24/3259/235032">Jonikas et al., 2009a</a>) is often used, followed by structure refinement to remove steric clashes and chain breaks</p></li>
</ol>
</li>
<li><p>12 papers that differ slightly in these steps</p>
<ul class="simple">
<li><p>ex. graph-based: 2d structure mapped to a graph</p></li>
<li><p>three open-source papers that yield all-atom structure</p>
<ul>
<li><p><strong>IsRNA1</strong>: <em>De Novo</em> Prediction and Blind Screening of RNA 3D Structures (<a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/acs.jctc.0c01148">zhang et al. 2021</a>)</p>
<ul>
<li><p>origina IsRNA (<a class="reference external" href="https://pubs.acs.org/doi/abs/10.1021/acs.jctc.7b01228">zhang &amp; chen, 2018</a>)</p></li>
</ul>
</li>
<li><p><strong>iFoldRNA</strong>: three-dimensional RNA structure prediction and folding (<a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/18579566/">sharma et al. 2008</a> )</p></li>
<li><p><strong>SimRNA</strong> (<a class="reference external" href="https://academic.oup.com/nar/article-abstract/44/7/e63/2467805">boiecki et al. 2016</a>)</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</section>
</section>
<section id="protein-structure-prediction-dna">
<h2><span class="section-number">1.3.3. </span>protein structure prediction (dna)<a class="headerlink" href="#protein-structure-prediction-dna" title="Link to this heading">#</a></h2>
<section id="protein-basics">
<h3><span class="section-number">1.3.3.1. </span>protein basics<a class="headerlink" href="#protein-basics" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p>the standard way to obtain the 3D structure of a protein is X-ray crystallography</p>
<ul>
<li><p>takes ~1 year &amp;  $120k to obtain the structure of a single protein through X-ray crystallography (<a class="reference external" href="https://fortune.com/2020/11/30/deepmind-protein-folding-breakthrough/">source</a>)</p></li>
<li><p>alternative: nuclear magnetic resonance (NMR) spectroscopy</p></li>
</ul>
</li>
<li><p>on average, a protein is composed of 300 amino acids (residues)</p>
<ul>
<li><p>21 amino acid types</p></li>
<li><p>the first residue is fixed</p></li>
</ul>
</li>
<li><p>proteins evolved via mutations (e.g. additions, substitutions)</p>
<ul>
<li><p>fitness selects the best one</p></li>
</ul>
</li>
</ul>
</section>
<section id="algorithm-basics">
<h3><span class="section-number">1.3.3.2. </span>algorithm basics<a class="headerlink" href="#algorithm-basics" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://www.sciencedirect.com/topics/medicine-and-dentistry/multiple-sequence-alignment">multiple-sequence-alignment</a> (MSA) - alignment of 3 or more amino acid (or nucleic acid) sequences, which show conserved regions within a protein family which are of structural and functional importance</p>
<ul>
<li><p>matrix is (n_proteins x n_amino_acids)</p></li>
<li><p>independent model (PSSM) - model likelihood for each column independently</p></li>
<li><p>pairwise sites (potts model) - model likelihood for pairs of positions</p></li>
</ul>
</li>
<li><p>prediction problems</p>
<ul>
<li><p>contact prediction - binary map for physical contacts in the final protein</p>
<ul>
<li><p>common to predict this, evaluation uses binary metrics</p></li>
</ul>
</li>
<li><p>protein folding - predict 3d positions of all amino acids in the output structure</p></li>
<li><p><a class="reference external" href="https://arxiv.org/abs/1906.08230">Evaluating Protein Transfer Learning with TAPE</a> - given embeddings, 5 tasks to measure downstream performance (rao et al. 2019)</p>
<ul>
<li><p>tasks: structure (SS + contact), evolutionary (homology), engineering (fluorescence + stability)</p></li>
</ul>
</li>
<li><p>future: interactions between molecules (e.g. protein-protein, environment, highly designed proteins)</p></li>
</ul>
</li>
</ul>
</section>
<section id="deep-learning-papers">
<h3><span class="section-number">1.3.3.3. </span>deep-learning papers<a class="headerlink" href="#deep-learning-papers" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://www.nature.com/articles/s41586-021-04184-w">De novo protein design by deep network hallucination | Nature</a> (anishchenko…baker, 2021)</p>
<ul>
<li><p>deep networks trained to predict native protein structures from their sequences can be inverted to design new proteins</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.nature.com/articles/s41586-021-03819-2">Highly accurate protein structure prediction with AlphaFold | Nature</a> (jumper, …, hassabis, 2021)</p>
<ul>
<li><p><a class="reference external" href="https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-021-03819-2/MediaObjects/41586_2021_3819_MOESM1_ESM.pdf">supp</a></p></li>
<li><p><a class="reference external" href="https://www.blopig.com/blog/2021/07/alphafold-2-is-here-whats-behind-the-structure-prediction-miracle/">best blog post</a> (<a class="reference external" href="https://towardsdatascience.com/unfolding-alphafold-683d576a54a3">other blog post</a>)</p></li>
<li><p>model overview</p>
<ul>
<li><p><img alt="alphafold" src="../../_images/alphafold.png" /></p></li>
</ul>
</li>
<li><p>preprocessing</p>
<ul>
<li><p>MSA (<span class="math notranslate nohighlight">\(N_{seq} \times N_{res}\)</span>)</p>
<ul>
<li><p>uses evolutionary information</p></li>
</ul>
</li>
<li><p>pair representation (<span class="math notranslate nohighlight">\(N_{res} \times N_{res}\)</span>)</p>
<ul>
<li><p>finding templates - find similar proteins to model “pairs of residues” - which residues are likely to interact with each other</p></li>
<li><p>residue is the same as an amino acid</p></li>
</ul>
</li>
</ul>
</li>
<li><p>evoformer</p>
<ul>
<li><p>uses attention on graph network</p></li>
<li><p>iterative</p></li>
</ul>
</li>
<li><p>structure model - converts msa/pair representations into set of (x,y,z) coordinates</p>
<ul>
<li><p>“invariant point attention” - invariance to translations and rotations</p></li>
<li><p>predicts a global frame (residue gas)</p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference external" href="https://proceedings.mlr.press/v139/rao21a.html">MSA Transformer</a> (rao et al. 2021) - predicts contact prediction</p>
<ul>
<li><p>train via masking</p></li>
<li><p>output (contact prediction) is just linear combination of attention heads finetuned on relatively little data</p></li>
<li><p>some architecture tricks</p>
<ul>
<li><p>attention is axial - applied to rows / columns rather than entire input</p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference external" href="https://arxiv.org/abs/2202.05146">EquiBind: Geometric Deep Learning for Drug Binding Structure Prediction</a></p></li>
</ul>
</section>
<section id="covid">
<h3><span class="section-number">1.3.3.4. </span>covid<a class="headerlink" href="#covid" title="Link to this heading">#</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4147684/">The Coronavirus Nucleocapsid Is a Multifunctional Protein</a> (mcbride et al. 2014)</p>
<ul>
<li><p>coronavirus nucleocapsid = N protein</p></li>
</ul>
</li>
</ul>
</section>
</section>
<section id="tips-using-viruses-for-treatment">
<h2><span class="section-number">1.3.4. </span>TIPs - using viruses for treatment<a class="headerlink" href="#tips-using-viruses-for-treatment" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p>videos</p>
<ul>
<li><p><a class="reference external" href="https://www.ted.com/talks/leor_weinberger_can_we_create_vaccines_that_mutate_and_spread?language=en#t-798841">Leor Weinberger: Can we create vaccines that mutate and spread? | TED Talk</a></p></li>
<li><p><a class="reference external" href="https://www.youtube.com/watch?v=Dh0RbiAf2jY">Engineering Viruses - Leor Weinberger - Gladstone Institutes 2016 Fall Symposium - YouTube</a></p></li>
</ul>
</li>
<li><p>2 issues</p>
<ul>
<li><p>mutation - viruses mutate, our drugs don’t</p></li>
<li><p>transmission - adherence / deployment - hard to give certain the drugs</p></li>
</ul>
</li>
<li><p>solution: use modified versions of viruses as treatment</p>
<ul>
<li><p><em>therapeutic interfering particles</em> or <em>TIPs</em> are engineered deletion mutants designed to piggyback on a virus and deprive the virus of replication material</p>
<ul>
<li><p>TIP is like a parasite of the virus - it clips off some of the virus DNA (via genetic engineering)</p>
<ul>
<li><p>it doesn’t contain the code for replication, just for getting into the cell</p></li>
<li><p>since it’s shorter, it’s made more efficiently - thus it outcompetes the virus</p></li>
</ul>
</li>
</ul>
</li>
<li><p>how do TIPs spread?</p>
<ul>
<li><p>mutations happen because the “copy machine” within a cell makes a particular mutation</p>
<ul>
<li><p>when a new virus comes along, it makes some of the mutated parts</p></li>
<li><p>TIPs can’t replicate, so they take some of the mutated parts made by the new virus</p></li>
<li><p>then, the TIP gets copied with the same mutation as the virus and this now spreads</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li><p>effect</p>
<ul>
<li><p>viral load will immediately be lower</p></li>
<li><p>superspreaders can spread treatment to others</p></li>
<li><p>can take before exposure as well (although harder to get approval for this)</p></li>
</ul>
</li>
<li><p>note: HIV mutates very fast so a TIP in a single person sees a lot of variants</p>
<ul>
<li><p>HIP also enters the genome (so corresponding TIP does this as well)</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0092867421013192?dgcid=coauthor">Identification of a therapeutic interfering particle—A single-dose SARS-CoV-2 antiviral intervention with a high barrier to resistance</a> (chaturvedi…weinberger, 2021)</p>
<ul>
<li><p>DIP = defective interfering particle = wild-type virus</p></li>
<li><p>single administration of TIP RNA inhibits SARS-CoV-2 sustainably in continuous cultures</p></li>
<li><p>in hamsters, both prophylactic and therapeutic intranasal administration of lipid-nanoparticle TIPs durably suppressed SARS-CoV-2 by 100-fold in the lungs, reduced pro-inflammatory cytokine expression, and prevented severe pulmonary edema</p></li>
<li><p>TIP consists 1k - 2k bases</p></li>
<li><p>hard to actually look at structure here (requires cryoEM)</p></li>
</ul>
</li>
</ul>
</section>
<section id="random-papers">
<h2><span class="section-number">1.3.5. </span>random papers<a class="headerlink" href="#random-papers" title="Link to this heading">#</a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://www.nature.com/articles/d41586-021-03697-8?utm_source=Nature+Briefing&amp;amp;utm_campaign=b0bda6ba48-briefing-dy-20211210&amp;amp;utm_medium=email&amp;amp;utm_term=0_c9dfd39373-b0bda6ba48-45837550">DeepMind AI tackles one of chemistry’s most valuable techniques</a></p></li>
<li><p><a class="reference external" href="https://www.nature.com/articles/d41586-021-03593-1?utm_source=Nature+Briefing&amp;amp;utm_campaign=b0bda6ba48-briefing-dy-20211210&amp;amp;utm_medium=email&amp;amp;utm_term=0_c9dfd39373-b0bda6ba48-45837550">DeepMind’s AI helps untangle the mathematics of knots</a></p></li>
</ul>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./notes/research_ovws"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="ovw_disentanglement.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title"><span class="section-number">1.2. </span>disentanglement</p>
      </div>
    </a>
    <a class="right-next"
       href="ovw_llms.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">1.4. </span>llms</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#overview">1.3.1. overview</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#vocabulary">1.3.1.1. vocabulary</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#data">1.3.1.2. data</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#code">1.3.1.3. code</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#rna-structure-prediction">1.3.2. rna structure prediction</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#rna-basics">1.3.2.1. rna basics</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#algorithms">1.3.2.2. algorithms</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#d-rna-structure-prediction">1.3.2.3. 3d rna structure prediction</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#protein-structure-prediction-dna">1.3.3. protein structure prediction (dna)</a><ul class="nav section-nav flex-column">
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#protein-basics">1.3.3.1. protein basics</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#algorithm-basics">1.3.3.2. algorithm basics</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#deep-learning-papers">1.3.3.3. deep-learning papers</a></li>
<li class="toc-h3 nav-item toc-entry"><a class="reference internal nav-link" href="#covid">1.3.3.4. covid</a></li>
</ul>
</li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#tips-using-viruses-for-treatment">1.3.4. TIPs - using viruses for treatment</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#random-papers">1.3.5. random papers</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Chandan Singh
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright None.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
<div class="extra_footer">
  <p>
Many of these images are taken from resources on the web.
</p>
</div>
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../../_static/scripts/bootstrap.js?digest=dfe6caa3a7d634c4db9b"></script>
<script src="../../_static/scripts/pydata-sphinx-theme.js?digest=dfe6caa3a7d634c4db9b"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>